lms/djangoapps/shoppingcart/models.py

```diff
@@ -1352,6 +1352,13 @@ def add_to_order(cls, order, course_id, mode_slug=CourseMode.DEFAULT_MODE_SLUG,
                         .format(order.user.email, course_id, order.id))
             raise AlreadyEnrolledInCourseException
 
+        course_modes = CourseMode.modes_for_course_dict(course_id)
+        # Check if the NO_ID_PROFESSIONAL mode is defined for the given course
+        # then use this mode for registration instead of default 'honor' mode to
+        # use the correct currency and amount for registration
+        if CourseMode.NO_ID_PROFESSIONAL_MODES[0] in course_modes:
+            mode_slug = CourseMode.NO_ID_PROFESSIONAL_MODES[0]
```

Comment:
This should be handled at a higher layer in `verify_student.views.create_order`.  The current version of that code retrieves the verified mode for the course; the new version should retrieve the first paid mode (same logic as when the user enters the payment/verification flow).  You can then pass that course mode slug as the `mode_slug` kwarg of this method.